src/common.test.cc

```diff
@@ -95,3 +95,35 @@ TEST(common, RemoveZeroProbabilityErrors) {
   ASSERT_EQ(flat.instructions[1].type, stim::DemInstructionType::DEM_ERROR);
   EXPECT_NEAR(flat.instructions[1].arg_data[0], 0.2, 1e-9);
 }
+
+// Helper function to compare the two methods.
+void assert_merged_probabilities_are_equal(double p1, double p2) {
+  // Method 1: Merge probabilities directly using the exclusive OR formula.
+  double merged_p_direct = p1 + p2 - 2 * p1 * p2;
+
+  // Method 2: Convert to likelihood costs, merge them, then convert back.
+  double cost1 = -1 * std::log(p1 / (1 - p1));
+  double cost2 = -1 * std::log(p2 / (1 - p2));
+  double merged_cost = common::merge_weights(-cost1, -cost2);
```

Comment:
There is no reason to flip the sign here. The merge_weights function takes as input two likelihood costs just as defined in the code here. The weight is just `log((1-p)/p)` which is identical to the `-log(p/(1-p))` (how it's implemented in the tesseract code).

```suggestion
  double merged_cost = common::merge_weights(cost1, cost2);
```

Reply:
Thank you. Done.